### PR TITLE
crude loading indicator

### PR DESCRIPTION
**Pull request by pawelkaczmarekuk**
### JIRA link (if applicable) ###

EM-1633

### Change description ###

Crude implementation of a loading indicator.

### CODE REVIEW

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.spec.ts [hunk 1]
```diff
@@ -54,14 +69,39 @@ describe('PdfJsWrapper', () => {
...
+    const documentLoadedSpy = spyOn(wrapper.documentLoaded, 'next').and.callThrough();
+    const documentLoadFailedSpy = spyOn(wrapper.documentLoadFailed, 'next').and.callThrough();
+    const loadingTask = Promise.reject(new Error('x'));
+
+    // hack out the pdf.js function
+    pdfjsLib.getDocument = () => loadingTask;
+
+    await wrapper.loadDocument({} as any);
+
+    expect(pdfViewerSpy).toHaveBeenCalledTimes(0);
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.spec.ts]: not.toHaveBeenCalled()

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts [hunk 2]
```diff
@@ -140,4 +144,18 @@ describe('PdfViewerComponent', () => {
...
+  });
+
+  it('on DocumentLoadProgress indicate document loading progress', () => {
+    mockWrapper.documentLoadProgress.next(new DocumentLoadProgress(10, 100));
+    expect(component.loadingDocumentProgress).toBe(10);
+    mockWrapper.documentLoadProgress.next(new DocumentLoadProgress(90, 100));
+    expect(component.loadingDocumentProgress).toBe(90);
+    mockWrapper.documentLoadProgress.next(new DocumentLoadProgress(200, 100));
+    expect(component.loadingDocumentProgress).toBe(100);
+  });
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts]: should these be separate tests

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts [hunk 3]
```diff
@@ -42,6 +45,11 @@ export class PdfViewerComponent implements AfterViewInit, OnChanges {
     this.pdfWrapper = this.pdfJsWrapperFactory.create(this.viewerContainer);
     this.pdfWrapper.currentPageChanged.subscribe(v => this.currentPageChanged.next(v));
     this.pdfWrapper.searchResults.subscribe(v => this.searchResults.next(v));
+    this.pdfWrapper.documentLoadInit.subscribe(() => this.onDocumentLoadInit());
+    this.pdfWrapper.documentLoadProgress.subscribe(v => this.onDocumentLoadProgress(v));
+    this.pdfWrapper.documentLoaded.subscribe(() => this.onDocumentLoaded());
+    this.pdfWrapper.documentLoadFailed.subscribe(() => this.onDocumentLoadFailed());
+    this.pdfWrapper.searchResults.subscribe(v => this.searchResults.next(v));
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts]: this looks like it's duplicated

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts (hunk 2 above)
Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts]: nah - let's not split hair...

**Pull request merged**